projects/igniteui-angular/src/lib/grids/common/pipes.ts

```diff
@@ -229,7 +229,9 @@ export class IgxGridRowPinningPipe implements PipeTransform {
 
         if (grid.hasPinnedRecords && isPinned) {
             const result = collection.filter(rec => grid.isRecordPinned(rec));
-            result.sort((rec1, rec2) => grid.pinRecordIndex(rec1) - grid.pinRecordIndex(rec2));
+            const isRecordNormalized = collection.some(rec => rec.hasOwnProperty('children') && rec.hasOwnProperty('data'));
```

Comment:
Base pinning pipe should not have tree grid specific code checks.
If data needs to be extracted then the pinRecordIndex can be overwritten in the Treegrid component.